### PR TITLE
187167707 teacher form submission

### PR DESCRIPTION
**Pull request by ArushC**
### [Pivotal Tracker Link][tracker]

[tracker]: https://www.pivotaltracker.com/story/show/your-story-id

## What this PR does:

Sends an email to the teacher (in addition to the admin) whenever they fill out a form. The new corresponding email template title is "Teacher Form Submission."

This pull request fixes|implements (pick one...) ______.

#### Who authored this PR?

Arush

### How should this PR be tested?

Look at the RSpec and Cucumber tests to confirm that the feature works as intended.

* Is there a deploy we can view?
* What do the specs/features test?
* Are there edge cases to watch out for?

#### Are there any complications to deploying this?

Run "rails db:seed".

### Checklist:

- [ ] Has this been deployed to a staging environment or reviewed by a customer?
- [ ] Tag someone for code review (either a coach / team member)
- [ ] I have renamed the branch to match PivotTracker's suggested one (necessary for BlueJay) (e.g. `michael/12345-add-new-feature` Any branch name will do as long as the story ID is there. You can use `git checkout -b [new-branch-name]`)

### CODE REVIEW

File: app/mailers/teacher_mailer.rb [hunk 1]
```diff
@@ -11,44 +11,57 @@ class TeacherMailer < ApplicationMailer
...
+    form_submission(teacher)
+  end
+
   private
   def liquid_assigns
     base_rules = {
       bjc_password: Rails.application.secrets[:bjc_password],
       piazza_password: Rails.application.secrets[:piazza_password],
-      reason: @reason
+      # TODO: Review if below two are needed, or can they be refractored?
```
Comment (fp456) [inline app/mailers/teacher_mailer.rb:62]: clean up: remove the comment or move the comment before the function declaration and add the note

File: app/controllers/email_templates_controller.rb [hunk 2]
```diff
@@ -27,11 +32,11 @@ def create
     end
     load_ordered_email_templates
 
-    if @email_template.save!
+    if @email_template.save
```
Comment (fp456) [inline app/controllers/email_templates_controller.rb:35]: Adding testing for if statement

File: app/controllers/email_templates_controller.rb [hunk 3]
```diff
@@ -27,11 +32,11 @@ def create
...
       flash[:success] = "Created #{@email_template.title} successfully."
       redirect_to email_templates_path
     else
-      flash[:alert] = "Failed to submit information :("
+      flash[:alert] = "Failed to submit information: " + @email_template.errors.full_messages.join(", ")
```
Comment (fp456) [inline app/controllers/email_templates_controller.rb:39]: Testing for the else statement

**Review by fp456 — APPROVED** [review]
Nice job overall! Just a couple tests that can be added and cleaning up some lines. For the cleanup I was probably nitpicking, but probably necessary as this will be legacy code in the future :o

**Review by cycomachead — APPROVED** [review]
Looks great!
